UnitTests/Text/TextFormatterTests.cs

```diff
@@ -1418,5 +1604,129 @@ public void String_Array_Is_Not_Always_Equal_ToRunes_Array (string text, int run
+		}
+
+		[Fact]
+		public void GetSumMaxCharWidth_With_Combining_Runes ()
+		{
+			var text = "Les Mise\u0328\u0301rables";
+			Assert.Equal (1, TextFormatter.GetSumMaxCharWidth (text, 1, 1));
+		}
+
+		[Fact]
+		public void GetSumMaxCharWidth_List_With_Combining_Runes ()
+		{
+			var text = new List<string> () { "Les Mis", "e\u0328\u0301", "rables" };
+			Assert.Equal (1, TextFormatter.GetSumMaxCharWidth (text, 1, 1));
+		}
+
+		[Theory]
+		[InlineData (14, 1, TextDirection.LeftRight_TopBottom)]
+		[InlineData (1, 14, TextDirection.TopBottom_LeftRight)]
+		public void CalcRect_With_Combining_Runes (int width, int height, TextDirection textDirection)
+		{
+			var text = "Les Mise\u0328\u0301rables";
+			Assert.Equal (new Rect (0, 0, width, height), TextFormatter.CalcRect (0, 0, text, textDirection));
+		}
+
+		[Theory, AutoInitShutdown]
+		[InlineData (14, 1, TextDirection.LeftRight_TopBottom, "Les Misęrables")]
+		[InlineData (1, 14, TextDirection.TopBottom_LeftRight, "L\ne\ns\n \nM\ni\ns\nę\nr\na\nb\nl\ne\ns")]
+		[InlineData (4, 4, TextDirection.TopBottom_LeftRight, @"
+LMre
+eias
+ssb 
+ ęl ")]
+		public void Draw_With_Combining_Runes (int width, int height, TextDirection textDirection, string expected)
+		{
+			var text = "Les Mise\u0328\u0301rables";
+
+			var tf = new TextFormatter ();
+			tf.Direction = textDirection;
+			tf.Text = text;
+
+			Assert.True (tf.WordWrap);
+			if (textDirection == TextDirection.LeftRight_TopBottom) {
+				Assert.Equal (new Size (width, height), tf.Size);
+			} else {
+				Assert.Equal (new Size (1, text.GetColumns ()), tf.Size);
+				tf.Size = new Size (width, height);
+			}
+			tf.Draw (new Rect (0, 0, width, height), new Attribute (ColorName.White, ColorName.Black), new Attribute (ColorName.Blue, ColorName.Black));
+			TestHelpers.AssertDriverContentsWithFrameAre (expected, output);
+		}
+
+		[Theory, AutoInitShutdown]
+		[InlineData (17, 1, TextDirection.LeftRight_TopBottom, 4, "This is a     Tab")]
+		[InlineData (1, 17, TextDirection.TopBottom_LeftRight, 4, "T\nh\ni\ns\n \ni\ns\n \na\n \n \n \n \n \nT\na\nb")]
+		[InlineData (13, 1, TextDirection.LeftRight_TopBottom, 0, "This is a Tab")]
+		[InlineData (1, 13, TextDirection.TopBottom_LeftRight, 0, "T\nh\ni\ns\n \ni\ns\n \na\n \nT\na\nb")]
+		public void TabWith_PreserveTrailingSpaces_False (int width, int height, TextDirection textDirection, int tabWidth, string expected)
+		{
+			var text = "This is a \tTab";
+			var tf = new TextFormatter ();
+			tf.Direction = textDirection;
+			tf.TabWidth = tabWidth;
+			tf.Text = text;
+
+			Assert.True (tf.WordWrap);
+			Assert.False (tf.PreserveTrailingSpaces);
+			Assert.Equal (new Size (width, height), tf.Size);
+			tf.Draw (new Rect (0, 0, width, height), new Attribute (ColorName.White, ColorName.Black), new Attribute (ColorName.Blue, ColorName.Black));
+			TestHelpers.AssertDriverContentsWithFrameAre (expected, output);
+		}
+
+		[Theory, AutoInitShutdown]
```

Comment:
Is it possible to refactor these tests to NOT use AutoInitShutdown?

IMO, TextFormatter should be able to operate completely independently of `Application`. 

If there is some reason it cannot, I consider that a bug, and we should either fix it in this PR or open another PR to do so in the future.

Reply:
I already added this commit which fix this issue https://github.com/gui-cs/Terminal.Gui/pull/3012/commits/4fe1392886048b470751eaaa6f327a9bc75a47de